src/java.base/share/classes/java/lang/StringCoding.java

```diff
@@ -85,9 +88,24 @@ public static boolean hasNegatives(byte[] ba, int off, int len) {
      *   bytes in the range. If there are negative bytes, the implementation must return
      *   a value that is less than or equal to the index of the first negative byte
      *   in the range.
+     *
+     * @param ba a byte array
+     * @param off the index of the first byte to start reading from
+     * @param len the total number of bytes to read
+     * @throws NullPointerException if {@code ba} is null
+     * @throws ArrayIndexOutOfBoundsException if the provided sub-range is
+     *         {@linkplain Preconditions#checkFromIndexSize(int, int, int, BiFunction) out of bounds}
      */
+    static int countPositives(byte[] ba, int off, int len) {
```

Comment:
If we name countPositives with parameter checking as countPositivesSB, this PR will have fewer changes.